api/post.go

```diff
@@ -728,6 +730,92 @@ func updateMentionCountAndForget(channelId, userId string) {
 	}()
 }
 
+func checkForOutOfChannelMentions(c *Context, post *model.Post, channel *model.Channel) {
+	// don't check for out of channel mentions in direct channels
+	if channel.Type == model.CHANNEL_DIRECT {
+		return
+	}
+
+	mentioned := getOutOfChannelMentions(post, channel.TeamId)
+	if len(mentioned) == 0 {
+		return
+	}
+
+	usernames := make([]string, len(mentioned))
+	for i, user := range mentioned {
+		usernames[i] = user.Username
+	}
+	sort.Strings(usernames)
+
+	var messageText string
```

Comment:
@hmhealey I haven't tried it yet but there's [support](https://github.com/nicksnyder/go-i18n#translation-files) for that.